src/play_controller.cpp

```diff
@@ -1196,6 +1198,85 @@ void play_controller::start_game()
+ */
+static void find_next_scenarios(const config& parent, std::set<std::string>& result) {
+	for(const auto& endlevel : parent.child_range("endlevel")) {
+		if(endlevel.has_attribute("next_scenario")) {
+			result.insert(endlevel["next_scenario"]);
+		}
+	}
+	for(const auto& cfg : parent.all_children_range()) {
+		find_next_scenarios(cfg.cfg, result);
+	}
+};
+
+void play_controller::check_next_scenario_is_known() {
+	// Which scenarios are reachable from the current one?
+	std::set<std::string> possible_next_scenarios;
+	possible_next_scenarios.insert(gamestate().gamedata_.next_scenario());
+
+	// Find all "endlevel" tags that could be triggered in events
+	config events;
+	gamestate().events_manager_->write_events(events);
+	find_next_scenarios(events, possible_next_scenarios);
+
+	// Are we looking for [scenario]id=, [multiplayer]id= or [test]id=?
+	const auto tagname = saved_game_.classification().get_tagname();
+
+	// Of the possible routes, work out which exist.
+	bool possible_this_is_the_last_scenario = false;
+	std::vector<std::string> known;
+	std::vector<std::string> unknown;
+	for(const auto& x : possible_next_scenarios) {
+		if(x.empty() || x == "null") {
+			possible_this_is_the_last_scenario = true;
+			LOG_NG << "This can be the last scenario\n";
+		} else if(game_config_.find_child(tagname, "id", x)) {
+			known.push_back(x);
+			LOG_NG << "Known next scenario '" << x << "'\n";
+		} else {
+			unknown.push_back(x);
+			ERR_NG << "Unknown next scenario '" << x << "'\n";
+		}
+	}
+
+	if(unknown.empty()) {
+		// everything's good
+		return;
+	}
+
+	std::string message;
+	if(known.empty() && !possible_this_is_the_last_scenario) {
+		// TRANSLATORS: This is an error that will hopefully only be seen by UMC authors and by players who have already said "okay" to a "loading saves from an old version might not work" dialog. It should be rare enough that there aren't separate strings for singular vs plural in the $unknown_list.
+		message = _("There will be an “unknown scenario” error instead of the next scenario. This is likely to be a bug in the campaign, or the result of loading an old saved game that’s incompatible with the current version of the campaign.\n\nThese scenarios are missing:\n\n$unknown_list|");
+	} else {
+		// TRANSLATORS: This is an error that will hopefully only be seen by UMC authors and by players who have already said "okay" to a "loading saves from an old version might not work" dialog. It should be rare enough that there aren't separate strings for singular vs plural in either of the lists.
+		message = _("Some of the ways to win this scenario will lead to an “unknown scenario” error instead of the next scenario. This is likely to be a bug in the campaign, or the result of loading an old saved game that’s incompatible with the current version of the campaign.\n\nThe following branches are expected to be ok:\n\n$known_list|\nThe following branches will lead to an error:\n\n$unknown_list|");
```

Comment:
I still need a better way to give these long lines to `gettext`.

Reply:
Ah, gettext supports string concatenation. Easier than I thought, not sure why I didn't try that before pushing the original version.

Reply:
By string concatenation do you mean `_("some text\n"  "more text\n)`?

Reply:
Yes.